# Itineraires with the same starting points
to see it rendered visit https://nbviewer.org/github/Universite-Gustave-Eiffel/geodev-logicout/blob/Thiago/notebooks/same_starting_point.ipynb

After the findings of the index_distance notebook, we will inspect the candidates for mutualisations that doesn't have the same id of the itineraire searched but that starts at the same point 

In [1]:
import sys  
sys.path.insert(0, '../scripts')
import numpy as np
import indexes, use_data, IsInclude
import pandas as pd
from shapely import wkt,MultiLineString
import ast
import warnings
warnings.filterwarnings('ignore')
import aux_plot

In [2]:
parameters = {
'radius':100000,
'buffer_hull':1000,
"type":1,
"area" : np.pi*100000**2
}


geo_df = use_data.create_gdf('simulations_reel_gdf.csv','start')
ranked_simulations = pd.read_csv("../data/raw/ranked_mutualisations.csv")

Let's see how many itinaires doesn't have any candidate (buffer around cheflieu= 100km)

In [3]:
ranked_simulations['mutualisations'].isnull().sum()

20

In [4]:
#We prepare the database to allow those to be count correctly and take only the entries that have valid candidates
ranked_simulations['mutualisations']=ranked_simulations['mutualisations'].replace(np.nan,'[]',regex=True)
ranked_simulations['mutualisations'] = ranked_simulations['mutualisations'].map(lambda x: ast.literal_eval(x))
ranked_simulations['n_candidates'] = ranked_simulations['mutualisations'].map(lambda x: len(x))
ranked_simulations=ranked_simulations[ranked_simulations['n_candidates'] != 0]

In [5]:
ranked_simulations['n_candidates'].describe()

count    1077.000000
mean       72.354689
std        55.692852
min         1.000000
25%        31.000000
50%        58.000000
75%       105.000000
max       236.000000
Name: n_candidates, dtype: float64

Now we will inspect how many of those candidates have an index of mutualisation = 0

In [6]:
ranked_simulations['candidates_0']=ranked_simulations['mutualisations'].map(lambda x: [y for y in x if y[5]==0])
ranked_simulations['n_candidates_0'] = ranked_simulations['candidates_0'].map(lambda x: len(x))
ranked_simulations_index_0 = ranked_simulations[ranked_simulations['n_candidates_0'] != 0]

In [7]:
ranked_simulations_index_0[['n_candidates','n_candidates_0']].describe()

n_candidates  n_candidates_0
count    216.000000      216.000000
mean      81.356481        9.324074
std       38.808507       13.412822
min       15.000000        2.000000
25%       58.000000        4.000000
50%       58.000000        4.000000
75%      107.000000        9.000000
max      144.000000       92.000000

These are the stats for all the 216 itineraires that have at least one candidate with the index = 0. We can inspect some itineraires too see how they looks. The original itineraire is shown in gray

In [14]:
ranked_simulations_index_0['id_simulation'].values

array([ 6783,  6786,  6787,  6788,  6789,  6790,  6792,  6793,  6885,
        7969,  7970,  7971,  7972,  7973,  7974,  6813,  6814,  6815,
        6838,  7964,  7965,  6818,  6820,  6840,  6841,  7962,  7963,
        7976,  7977,  7978,  7983,  7984, 13899,  6886,  6888,  6889,
        6896,  7986,  7988,  7989,  7990,  6923,  6924,  6925,  6926,
        6927,  8002,  8003,  8008,  8009,  8010,  7755,  7756,  7757,
        7758,  7766,  7767,  8007,  8013,  8014,  7807,  8004,  8559,
        8565,  8566,  8567,  9251,  9298,  9257,  9258,  9259,  9268,
        9269,  9271,  9273,  9274,  9275,  9277,  9280,  9281,  9284,
        9287,  9295,  9296,  9297,  9299,  9302,  9306, 10683, 10684,
       10719, 10725, 10728, 10729, 10732, 10860, 10861, 10863, 10864,
       10865, 10867, 10868, 10872, 11127, 11169,  9313,  9314,  9322,
        9326,  9327,  9328,  9330,  9331,  9334,  9339,  9354,  9355,
        9366,  9368,  9371,  9384,  9388,  9389,  9393,  9399,  9404,
        9405,  9406,

In [15]:
table1, map1 = aux_plot.plot_mutualisations_zero_index(6783,geo_df,**parameters)
table2, map2 = aux_plot.plot_mutualisations_zero_index(9398,geo_df,**parameters)
table3, map3 = aux_plot.plot_mutualisations_zero_index(10739,geo_df,**parameters)



In [16]:
map1

In [17]:
map2

In [18]:
map3

In [13]:
#ranked_simulations['multilines'] = ranked_simulations['mutualisations'].map(lambda x: MultiLineString(x))